### ISSUE TITLE
2.0 - Deleted media files are not removed.

### ISSUE THREAD

**Issue by makitso** [fisharebest/webtrees#1815]
Go to an individual / media.
Select a media file and delete it.  However, this does not remove it from the webtrees database or file system.
Go to Manage Media and find the file
Delete the file
The media object is removed from the webtrees database but still resides in /media and the Thumbnail-cache.

**Comment by makitso** [comment]
The point is that once I delete the media file, I should be able to add the media item again.  However, in doing so, the system supplies its own file name rather than the one I requested.  I have looked in the database and file system and the original file was deleted. so am at a loss.

Also, IHMO, a  user should be able to delete a media item from the individual age and have it completely deleted.  One should not have to go to the Manage Media page and delete it -- 2 cents.

**Comment by makitso** [comment]
Aside from the original purpose of this issue, I have a suggestion to help users manage media.
I understand the webtrees concepts of media objects and links.
1.  On the individual Page / Media you can have the option to Delete the Media Object.  Perhaps it should read Media Link?

![image](https://user-images.githubusercontent.com/5147603/43913082-f1045882-9bb8-11e8-9f5e-01d5bbf9857c.png)

2.  In #1, if you delete the media object it should check to see if there are more than one link to that object.  If there is only one link then it should delete the link "and" the object.

**Comment by fisharebest** [comment]
> The point is that once I delete the media file, I should be able to add the media item again. However, in doing so, the system supplies its own file name rather than the one I requested. I have looked in the database and file system and the original file was deleted. so am at a loss.

When you delete a media object, the corresponding media file is **not** removed from disk.

Thus if you try to upload another file with the same name, webtrees will choose a new name to prevent you from overwriting the existing one.

We probably need the option to delete files at the same time as the media objects.
At present, the only way to do this is to delete "unused media files" from the control panel.

**Comment by makitso** [comment]
I would also update the individual/media page so that when it says "Media object"  to Media object link"

**Comment by wkitty42** [comment]
On 08/09/2018 05:18 PM, Greg Roach wrote:
> When you delete a media object, the corresponding media file is *not*
> removed from disk. >
> Thus if you try to upload another file with the same name, webtrees will
> choose a new name to prevent you from overwriting the existing one. >
> We probably need the option to delete files at the same time as the media
> objects. At present, the only way to do this is to delete "unused media
> files" from the control panel.


or maybe rewording the task... if someone says "delete" when working with files 
and links, they're likely meaning to delete both... i'm kinda of the same mind 
as rob...

so i was thinking, reword so that you "remove media link" and "delete media 
file"... deleting would remove the link and then delete the files from the drive 
and the cache... be explicit in the wording...

2 centavos from the peanut gallery...
